keras_cv/applications/object_detection/retina_net/__internal__/layers/decode_predictions.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+
+from keras_cv import layers
+from keras_cv.applications.object_detection.retina_net.__internal__ import utils
+
+
+class DecodePredictions(tf.keras.layers.Layer):
+    """A Keras layer that decodes predictions of the RetinaNet model.
+
+    Attributes:
+      num_classes: Number of classes in the dataset.
+      confidence_threshold: Minimum class probability, below which detections are
+        pruned.
+      nms_iou_threshold: IOU threshold for the NMS operation.
+      max_detections_per_class: Maximum number of detections to retain per class.
+      max_detections: Maximum number of detections to retain across all classes.
+      box_variance: The scaling factors used to scale the bounding box predictions.
+    """
+
+    def __init__(
+        self,
+        num_classes,
+        bounding_box_format,
+        confidence_threshold=0.05,
+        iou_threshold=0.5,
+        max_detections_per_class=100,
```

Comment:
not sure if those default param are generic enough, or should be changed based on different dataset. If we would like to share this between different models, we might consider make them required parameters.

Reply:
lets save this for the PR where we expose this to users